trio/_highlevel_open_unix_stream.py

```diff
@@ -36,12 +36,13 @@ def close_on_error(obj):
     Raises:
       OSError: If the socket file could not be connected to.
       RuntimeError: If AF_UNIX sockets are not supported.
+      TypeError: if filename is not str or bytes.
     """
     if not has_unix:
         raise RuntimeError("Unix sockets are not supported on this platform")
 
-    if filename is None:
-        raise ValueError("Filename cannot be None")
+    if not isinstance(filename, (str, trio.Path, bytes)):
+        raise TypeError("Filename must be str, trio.Path or bytes")
```

Comment:
This isn't *quite* correct... really the criterion is "is the object a valid path". And starting in Python 3.6, the proper way to check that is to use the [fspath protocol](https://www.python.org/dev/peps/pep-0519/). So if we want to do this check, then the right way to do it would be something like:

```python
filename = os.fspath(filename)
```

This automatically handles all "path-like" objects, including new ones defined in third-party libraries that we've never heard of. And if the user passes in something inappropriate, it gives a good error message:

```python
>>> os.fspath(4)                                                                      
TypeError: expected str, bytes or os.PathLike object, not int
```

Annoyingly, though, we still support Python 3.5 (though not for long, see #1396), so we can't just use `os.fspath`. Instead, use our compatibility wrapper: `trio._util.fspath`

Reply:
Thank you for the information, I didn't know the function `os.fspath`. To be honest, when I made the change, I told myself that a `pathlib.Path` object would also be valid but I was "comforted" by the fact that the tests only took into account `trio.Path` 😛
I changed the check